### PR TITLE
refactor: add a `Consent` proto data type

### LINKED ISSUE #588: Add a `Consent` proto type
When the providers submit consent, the provider will use `Consent` proto type. So, add a `Consent` proto type and put the `Certificate` field like following code: 
```proto
message Consent {
     Certificate certificate = 1;
}
```

### PR DESCRIPTION
**Pull request by inchori**
Closes #588

### CODE REVIEW

File: x/datadeal/keeper/grpc_query_deal_test.go [hunk 1]
```diff
@@ -64,86 +64,94 @@ func (suite *queryDealTestSuite) TestQueryDeals() {
 }
 
 func (suite *queryDealTestSuite) TestQueryCert() {
```
Comment (H4NLee) [inline x/datadeal/keeper/grpc_query_deal_test.go:66]:
```suggestion
func (suite *queryDealTestSuite) TestQueryConsent() {
```

File: x/datadeal/keeper/grpc_query_deal_test.go [hunk 2]
```diff
@@ -64,86 +64,94 @@ func (suite *queryDealTestSuite) TestQueryDeals() {
...
 	}
-	res, err := suite.DataDealKeeper.Certificate(sdk.WrapSDKContext(suite.Ctx), req)
+	res, err := suite.DataDealKeeper.Consent(sdk.WrapSDKContext(suite.Ctx), req)
 	suite.Require().NoError(err)
 	suite.Require().NotNil(res)
-	suite.Require().Equal(cert, res.Certificate)
+	suite.Require().Equal(consent, res.Consent)
 }
 
 func (suite *queryDealTestSuite) TestQueryCerts() {
```
Comment (H4NLee) [inline x/datadeal/keeper/grpc_query_deal_test.go:93]:
```suggestion
func (suite *queryDealTestSuite) TestQueryConsents() {
```

**Review by H4NLee — APPROVED** [review]
left minor comments. lgtm

File: x/datadeal/keeper/grpc_query_deal_test.go (hunk 2 above)
Comment (inchori) [reply x/datadeal/keeper/grpc_query_deal_test.go]: Fixed in [bdffe6c](https://github.com/medibloc/panacea-core/pull/590/commits/bdffe6c49e8bdcf116033400a534e5049ef1c3d3).

**Review by youngjoon-lee — APPROVED** [review]
lgtm

**Review by gyuguen — APPROVED** [review]
LGTM

**Review by H4NLee — APPROVED** [review]
LGTM

**Review by audtlr24 — APPROVED** [review]
lgtm

**Pull request merged**